src/mdio/segy/_workers.py

```diff
@@ -121,23 +122,31 @@ def trace_worker(  # noqa: PLR0913
     zarr_config.set({"threading.max_workers": 1})
 
     live_trace_indexes = local_grid_map[not_null].tolist()
-    traces = segy_file.trace[live_trace_indexes]
 
     header_key = "headers"
+    raw_header_key = "raw_headers"
+
+    # Used to disable the reverse transforms if we aren't going to write the raw headers
+    do_reverse_transforms = False
 
     # Get subset of the dataset that has not yet been saved
     # The headers might not be present in the dataset
     worker_variables = [data_variable_name]
     if header_key in dataset.data_vars:  # Keeping the `if` here to allow for more worker configurations
         worker_variables.append(header_key)
+    if raw_header_key in dataset.data_vars:
+        do_reverse_transforms = True
+        worker_variables.append(raw_header_key)
 
+    raw_headers, transformed_headers, traces = get_header_raw_and_transformed(
```

Comment:
```suggestion
    header_pipeline = deepcopy(segy_file.accessors.header_decode_pipeline)
    segy_file.accessors.header_decode_pipeline.transforms = []
```

Reversing transforms is not needed. you can extract the header transform pipeline and defer its application later. The above snipped I posted will skip all transforms, and you can write that out as `raw_headers`. Then `header_pipeline.apply(...)` will generate the interpreted versions and you can write them out as `headers. This way the `_disaster_recovery_wrapper.py` and all its tests can also be removed.  would still recommend keeping the tests that use a mock segy file of course.